### ISSUE TITLE
Incorrect gradients when using consts

### ISSUE THREAD

**Issue by andrewfurey21** [tinygrad/tinygrad#9597]
I have an issue where the gradients are being calculated incorrectly, I think because tinygrad sees `w1` and `w2` (in the example below) as indentical. 

```python
from tinygrad import Tensor
value1 = value2 = 3.0
w1 = Tensor(value1, requires_grad=True)
w2 = Tensor(value2, requires_grad=True)
out = w1.add(w2)
out.backward()
print(w1.grad.numpy(), w2.grad.numpy()) # gradients should be 1.0, not 2.0
```
I found the problem when adding two `Tensor.ones`.

I'm on the latest commit (`c011751b`), as of publishing this issue. I'm on python `3.10.12`. If `value1` or `value2` differ, there isn't a problem. Even if their types differ, there isn't a problem. Just when they are exactly the same. This works in tinygrad `0.10.0`. I ran a `git bisect` and the first bad commit was `b4bf6a7`

## Workaround

I traced down the problem and commented out a line`UOpMetaClass` in `ops.py` where checking the cache happens, but it fails a bunch of other tests, so not a solution.

```python
class UOpMetaClass(type):
  ucache:dict[tuple, weakref.ReferenceType[UOp]] = {}
  def __call__(cls, op:Ops, dtype:DType=dtypes.void, src:tuple[UOp,...]=tuple(), arg:Any=None, _buffer:Buffer|None=None):
    # if (wret:=UOpMetaClass.ucache.get(key:=(op, dtype, src, arg), None)) is not None and (ret:=wret()) is not None: return ret
    key=(op, dtype, src, arg)
    UOpMetaClass.ucache[key] = ref = weakref.ref(created:=super().__call__(*key))
    for s in src: s.children.add(ref)
    # NOTE: this will soon be set by Tensor once we remove function.py
    if (metadata:=_METADATA.get()) is not None: all_metadata[created] = metadata
    # NOTE: this value is set by pickle when pickling a realized tensor
    if _buffer is not None:
      assert op is Ops.BUFFER, f"trying to set Buffer {_buffer} for {op}"
      buffers[created] = _buffer
    return created
```
